File: trunk/server/src/main/java/org/ngame/protocol/notify/CardGroupNotify.java
/**
 * 资源更新
 */
package org.ngame.protocol.notify;

import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;
import lombok.Data;

import java.util.List;

import org.ngame.protocol.BaseResponse;
import org.ngame.protocol.annotation.Proto;
import org.ngame.protocol.domain.CardGroupInfo;

/**
 *
 * @author csf
 */
@Data
@Proto(value = -30, description = "套牌推送")
public class CardGroupNotify extends BaseResponse
{
   @Protobuf(description = "玩家id")
   private String pid;
   @Protobuf(description = "套牌组")
   private List<CardGroupInfo> cardGroup;
}

File: trunk/server/net/Protocols/CardGroupNotify.cs
using ProtoBuf;
using System.Collections.Generic;
using System;

namespace protocol
{
    /// <summary>
    /// 套牌推送 协议:-30
    /// </summary>
	[Proto(value=-30,description="套牌推送")]
	[ProtoContract]
	public class CardGroupNotify
	{
        /// <summary>
        ///  玩家id
        /// </summary>
		[ProtoMember(3, IsRequired = false)]
		public string pid;
        /// <summary>
        ///  套牌组
        /// </summary>
		[ProtoMember(4, IsRequired = false)]
		public List<CardGroupInfo> cardGroup;
        /// <summary>
        ///  是否成功
        /// </summary>
		[ProtoMember(1, IsRequired = false)]
		public bool success;
        /// <summary>
        ///  错误消息
        /// </summary>
		[ProtoMember(2, IsRequired = false)]
		public string info;

	}
}